### PR TITLE
feat(integrations): Add copper integration

### PR DESCRIPTION
**Pull request by dannylwe**

### CODE REVIEW

**Review by bodinsamuel — APPROVED** [review]

File: docs-v2/integrations/all/copper.mdx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+-   [OAuth related docs](https://developer.copper.com/introduction/oauth/quickstart.html)
+-   [List of OAuth scopes](https://developer.copper.com/introduction/oauth/index.html#7-scopes)
+-   [Copper API docs](https://developer.copper.com/introduction/requests.html#headers)
+-   [API rate liming](https://developer.copper.com/introduction/requests.html#rate-limits)
+
+<Tip>Need help getting started? Get help in the [community](https://nango.dev/slack).</Tip>
+
+## API gotchas
+
+- Copper offers both OAuth and API_KEY authentication modes. To use OAuth2.0 with Copper, you must first register your application.
```
Comment (khaliqgant) [inline docs-v2/integrations/all/copper.mdx]: Shouldn't we also add support for the API key auth as well?

File: docs-v2/integrations/all/copper.mdx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+## API gotchas
+
+- Copper offers both OAuth and API_KEY authentication modes. To use OAuth2.0 with Copper, you must first register your application. 
+To register your application, [contact](https://developer.copper.com/introduction/oauth/index.html#:~:text=To%20register%20your,all%20authorization%20flows.) copper.
+
+- You will need to provide the name and purpose of your application as well as a URL to an HTTPS endpoint that will handle the secure callback.
+You will receive two credentials, client_id and client_secret, required for all authorization flows.
+
+- The Dev API uses token based authentication. You have to include the token in the header of every request, along with the email address of the user who generated the token.
```
Comment (khaliqgant) [inline docs-v2/integrations/all/copper.mdx]: @dannylwe where in their documentation is this specified? Shouldn't we inject this for the user if this is the case?

File: packages/shared/providers.yaml [hunk 3]
```diff
@@ -1489,6 +1489,25 @@ codeclimate:
...
             description: The API key for your Code Climate account
 
+copper:
+    display_name: Copper
+    categories:
+        - crm
+    auth_mode: OAUTH2
+    proxy:
+        base_url: https://api.copper.com/developer_api/
+    docs: https://docs.nango.dev/integrations/all/copper
```
Comment (hassan254-prog) [inline packages/shared/providers.yaml]: It might be helpful to include the docs link at the end.

**Comment by hassan254-prog** [comment]
@khaliqgant, should we also start documenting for `connect-ui` now to prevent the need to backfill these docs in the future?

**Comment by khaliqgant** [comment]
> @khaliqgant, should we also start documenting for `connect-ui` now to prevent the need to backfill these docs in the future?

Yes, absolutely. Thanks for bringing this up.

File: docs-v2/integrations/all/copper.mdx (hunk 2 above)
Comment (dannylwe) [reply docs-v2/integrations/all/copper.mdx]: @khaliqgant Makes sense. This should be injected for the user

**Review by khaliqgant — CHANGES_REQUESTED** [review]
Changes requested inline

File: docs-v2/integrations/all/copper.mdx (hunk 1 above)
Comment (dannylwe) [reply docs-v2/integrations/all/copper.mdx]: This has been added

**Review by hassan254-prog — APPROVED** [review]

**Comment by bodinsamuel** [comment]
is it okay to merge?

File: docs-v2/integrations/all/copper.mdx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+- You will need to provide the name and purpose of your application as well as a URL to an HTTPS endpoint that will handle the secure callback.
+You will receive two credentials, client_id and client_secret, required for all authorization flows.
+
+- To generate an API token, in the Copper web app navigate to `System settings` > `Integrations` > `API Keys` and click the `GENERATE API KEY` button.
+
+<Note>Add Getting Started links and Gotchas by [editing this page](https://github.com/nangohq/nango/tree/master/docs-v2/integrations/all/copper.mdx)</Note>
+
+<Card title="Connect to copper-api-key" icon="link" href="/integrations/all/copper-api-key/connect" horizontal>
+  Guide to connect to Asana-Scim using Nango Connect.
```
Comment (khaliqgant) [inline docs-v2/integrations/all/copper.mdx]:
```suggestion
  Guide to connect to Copper API Key using Nango Connect.
```

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**